rasterio/rio/features.py

```diff
@@ -34,19 +34,22 @@
 @use_rs_opt
 @geojson_type_feature_opt(True)
 @geojson_type_bbox_opt(False)
-@click.option('--bands/--mask', default=True,
+@click.option('--band/--mask', default=True,
               help="Extract shapes from one of the dataset bands or from "
                    "its nodata mask")
-@click.option('--bidx', type=int, default=1,
+@click.option('--bidx', 'bandidx', type=int, default=None,
               help="Index of the source band")
 @click.option('--sampling', type=int, default=1,
               help="Inverse of the sampling fraction")
 @click.option('--with-nodata/--without-nodata', default=False,
               help="Include or do not include (the default) nodata regions.")
+@click.option('--as-mask/--not-as-mask', default=False,
+              help="Interpret a band as a mask and output only one class of "
+                   "valid data shapes.")
 @click.pass_context
 def shapes(
         ctx, input, precision, indent, compact, projection, sequence,
-        use_rs, geojson_type, bands, bidx, sampling, with_nodata):
+        use_rs, geojson_type, band, bandidx, sampling, with_nodata, as_mask):
     """Writes features of a dataset out as GeoJSON. It's intended for
```

Comment:
@sgillies Can you please expand the docstring to give a bit more explanation to usage of the parameters above, particularly how they should best be combined?